File: 191. Number of 1 Bits/number-of-1-bits.c
// 191. Number of 1 Bits (5/8/54369)
// Runtime: 3 ms (40.72%) Memory: 5.43 MB (38.83%) 

int hammingWeight (uint32_t n) {
    uint8_t count = 0;
    
    while (n) {
        n = n & (n - 1);
        count++;
    }
    
    return count;
}

File: 191. Number of 1 Bits/number-of-1-bits.ts
// 191. Number of 1 Bits (5/20/54369)
// Runtime: 109 ms (33.99%) Memory: 44.26 MB (87.93%) 

function hammingWeight(n: number): number {
    let count = 0;
    
    while (n) {
        count++;
        n = n & (n - 1);
    }
    
    return count;
};
